### PR TITLE
Validate and Format there is no space in the end of name or id.

### PR DESCRIPTION
**Pull request by daryakoval**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/40322

## Description
Validate and Format there is no space in the end of name or id in content items.

## Must have
- [x] Tests
- [x] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/common/errors.py [hunk 1]
```diff
@@ -1800,3 +1802,13 @@ def suggest_fix_field_name(field_name, pack_prefix):
     @error_code_decorator
     def entity_name_contains_excluded_word(entity_name: str, excluded_words: List[str]):
         return f'Entity {entity_name} should not contain one of {excluded_words} in its name. Please remove.'
+
+    @staticmethod
+    @error_code_decorator
+    def spaces_in_the_end_of_id(item_id: str):
+        return f'Content item id {item_id} should not contain spaces in the end. Please remove.'
```
Comment (dantavori) [inline demisto_sdk/commands/common/errors.py:1809]:
```suggestion
        return f'Content item id "{item_id}" should not have trailing spaces. Please remove.'
```

File: demisto_sdk/commands/common/errors.py [hunk 2]
```diff
@@ -1800,3 +1802,13 @@ def suggest_fix_field_name(field_name, pack_prefix):
...
+
+    @staticmethod
+    @error_code_decorator
+    def spaces_in_the_end_of_name(name: str):
+        return f'Content item name {name} should not contain spaces in the end. Please remove.'
```
Comment (dantavori) [inline demisto_sdk/commands/common/errors.py:1814]:
```suggestion
        return f'Content item name "{name}" should not have trailing spaces. Please remove.'
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,8 @@
 * Fixed an issue where yml files with '!reference' failed to load properly.
 * Fixed an issue when `View Integration Documentation` button was added twice during the download and re-upload.
 * Fixed an issue when `(Partner Contribution)` was added twice to the display name during the download and re-upload.
+* Added to **validate** command a validation that a content item's id and name will not end with spaces.
+* The **format** command will now remove whitespaces at the end of content item's id and name.
```
Comment (dantavori) [inline CHANGELOG.md:6]:
```suggestion
* The **format** command will now remove trailing whitespaces from content items' id and name fields.
```

File: demisto_sdk/commands/format/update_generic_json.py [hunk 4]
```diff
@@ -99,3 +100,18 @@ def update_id(self, field='name') -> None:
...
+        """Updates the id and name of the json to have no spaces on its end
+                """
+        updated_id = {}
+        if self.verbose:
+            click.echo('Updating YML ID and name to be without spaces at the end')
+
+        self.data['name'] = self.data['name'].rstrip()
+
+        if is_uuid(self.data['id']):
+            updated_id[self.data['id']] = self.data['id'].rstrip()
```
Comment (dantavori) [inline demisto_sdk/commands/format/update_generic_json.py:114]: what are these lines for?

File: demisto_sdk/commands/format/update_generic_yml.py [hunk 5]
```diff
@@ -243,3 +244,18 @@ def get_test_playbooks_configuration(test_playbooks: List, content_item_id: str,
...
+        """Updates the id and name of the YML to have no spaces on its end
+        """
+        updated_id = {}
+        if self.verbose:
+            click.echo('Updating YML ID and name to be without spaces at the end')
+
+        self.data['name'] = self.data['name'].rstrip()
+
+        if is_uuid(self.id_and_version_location['id']):
+            updated_id[self.id_and_version_location['id']] = self.id_and_version_location['id'].rstrip()
```
Comment (dantavori) [inline demisto_sdk/commands/format/update_generic_yml.py:258]: same question

**Review by dantavori — CHANGES_REQUESTED** [review]
looks good to me 👌
1. why not checking leading spaces as well?
2. let's make sure on sunday we want to allow ignoring these errors

File: demisto_sdk/commands/format/update_generic_json.py (hunk 4 above)
Comment (daryakoval) [reply demisto_sdk/commands/format/update_generic_json.py:114]: When you create content item at the UI, and then download it, id of this content item will be uuid - some hash string. If it happened, we will rename it, and then we should update all other items that uses it that id changed. 

Now when you asked I recognized that this is not necessary here.

Comment (daryakoval) [reply demisto_sdk/commands/format/update_generic_json.py:114]: We don't need it here.

File: demisto_sdk/commands/format/update_generic_yml.py (hunk 5 above)
Comment (daryakoval) [reply demisto_sdk/commands/format/update_generic_yml.py:258]: We don't need it here.

**Comment by daryakoval** [comment]
Waiting for https://github.com/demisto/content/pull/14465 to merge.

**Review by dantavori — APPROVED** [review]

**Pull request merged**